### PR TITLE
fix(compatibility): add missing mat- selectors

### PR DESCRIPTION
**Pull request by andrewseguin**

### CODE REVIEW

File: src/lib/input/input-container.html [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <div class="md-input-wrapper">
   <div class="md-input-table">
-    <div class="md-input-prefix"><ng-content select="[mdPrefix], [md-prefix]"></ng-content></div>
+    <div class="md-input-prefix">
+      <ng-content select="[mdPrefix], [matPrefix], [md-prefix], [mat-prefix]"></ng-content>
```
Comment (mmalerba) [inline src/lib/input/input-container.html]: don't add [mat-prefix], leave TODO to kill [md-prefix]

File: src/lib/input/input-container.html [hunk 2]
```diff
@@ -19,7 +21,9 @@
       </label>
     </div>
 
-    <div class="md-input-suffix"><ng-content select="[mdSuffix], [md-suffix]"></ng-content></div>
+    <div class="md-input-suffix">
+      <ng-content select="[mdSuffix], [matSuffix], [md-suffix], [mat-suffix]"></ng-content>
```
Comment (mmalerba) [inline src/lib/input/input-container.html]: no [mat-suffix], TODO kill [md-suffix]

File: src/lib/menu/menu-trigger.ts [hunk 3]
```diff
@@ -32,7 +32,8 @@ import {MenuPositionX, MenuPositionY} from './menu-positions';
  * responsible for toggling the display of the provided menu instance.
  */
 @Directive({
-  selector: '[md-menu-trigger-for], [mat-menu-trigger-for], [mdMenuTriggerFor]',
+  selector: `[md-menu-trigger-for], [mat-menu-trigger-for],
```
Comment (mmalerba) [inline src/lib/menu/menu-trigger.ts]: TODO kill kebab versions

**Comment by mmalerba** [comment]
tests seem unhappy for some reason

File: src/lib/input/input-container.html [hunk 4]
```diff
@@ -30,6 +36,6 @@
           [class.md-warn]="dividerColor == 'warn'"></span>
   </div>
 
-  <div *ngIf="hintLabel != ''" [attr.id]="_hintLabelId" class="md-hint">{{hintLabel}}</div>
+  <div *ngIf="hintLabel != ''" [attr.id]="_hintLabelId" class="md-hint, mat-hint">{{hintLabel}}</div>
```
Comment (kara) [inline src/lib/input/input-container.html]: Shouldn't `md-hint` be replaced with `mat-hint` rather than appended? Either way, comma should be removed.

Comment (kara) [inline src/lib/input/input-container.html]: I think the comma might be what's breaking the tests

Comment (andrewseguin) [inline src/lib/input/input-container.html]: Oh bugger I totally put that in the wrong spot, it should have been in the select attribute of the ng-content below it

**Pull request merged**